### PR TITLE
Add spork for staking on segwit

### PR DESCRIPTION
**Pull request by wqking**
This will deprecate the PR "Fix staking on bech32 #122"

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -4065,7 +4071,16 @@ bool CheckBlock(const CBlock& block, CValidationState& state, bool fCheckPOW, bo
 
     if (block.IsProofOfStake()) {
         // Coinbase output should be empty if proof-of-stake block
-        if (block.vtx[0].vout.size() != 1 || !block.vtx[0].vout[0].IsEmpty())
+        int commitpos = GetWitnessCommitmentIndex(block);
+        if (! hasEnabledStakingOnSegWit()) {
```
Comment (meyer9) [inline src/main.cpp]: remove space after `!` and just use `IsSporkActive(SPORK_19_STAKING_ON_SEGWIT)`. Also possibly needs to be moved into ContextualCheckBlock?

File: src/main.h [hunk 2]
```diff
@@ -659,4 +659,6 @@ struct CBlockTemplate {
 int64_t GetVirtualTransactionSize(const CTransaction& tx);
 int64_t GetVirtualTransactionSize(int64_t nCost);
 
+bool hasEnabledStakingOnSegWit();
```
Comment (meyer9) [inline src/main.h]: delete this

File: src/primitives/block.cpp [hunk 3]
```diff
@@ -207,12 +206,29 @@ bool CBlock::SignBlock(const CKeyStore& keystore)
             if (!keystore.GetKey(keyID, key))
                 return false;
 
-            //vector<unsigned char> vchSig;
             if (!key.Sign(GetHash(), vchBlockSig))
                  return false;
 
             return true;
         }
+        else if(whichType == TX_WITNESS_V0_KEYHASH)
```
Comment (meyer9) [inline src/primitives/block.cpp:214]: indentation incorrect

File: src/primitives/block.cpp [hunk 4]
```diff
@@ -259,6 +275,31 @@ bool CBlock::CheckBlockSignature() const
 
         return pubkey.Verify(GetHash(), vchBlockSig);
 
+    }
```
Comment (meyer9) [inline src/primitives/block.cpp:276]: indentation

**Review by meyer9 — CHANGES_REQUESTED** [review]

File: src/primitives/block.cpp (hunk 4 above)
Comment (wqking) [reply src/primitives/block.cpp:276]: I can't see the indention wrong. After I rewrote the indention to space (previously it was space too), git doesn't show any diff.
I removed the blank line at line 277.

File: src/primitives/block.cpp (hunk 3 above)
Comment (wqking) [reply src/primitives/block.cpp:214]: I can't see the indention wrong. After I rewrote the indention to space (previously it was space too), git doesn't show any diff. Please check again.

**Comment by meyer9** [comment]
I think there's a setting in git to ignore whitespace changes. Try turning that off.

**Pull request merged**